### PR TITLE
Filter out non-standard channels from `channelInfo` API

### PR DESCRIPTION
**Pull request by btcontract**
This change makes it possible for a non-standard channel to reply with `CommandFailure` which in turn will be filtered out in `channelsInfo` method.

The reason is that currently hosted channels have to return something to make the whole API call succeed and whatever they return would break external tools which only expect standard channel formats.

### CODE REVIEW

**Review by t-bast — COMMENTED** [review]
This needs a test (and probably a squash/rebase with a clean commit message), otherwise LGTM.

**Comment by btcontract** [comment]
Sigh, looks like I've done something wrong again with squashing, maybe we can do without it?

**Comment by t-bast** [comment]
> Sigh, looks like I've done something wrong again with squashing, maybe we can do without it?

Sure, no worries.

It's really worth a bit of time to learn to master git, it's not that hard and really helps productivity, I can only highly recommend it!

**Review by t-bast — COMMENTED** [review]
A couple of nits

File: eclair-core/src/test/scala/fr/acinq/eclair/EclairImplSpec.scala [hunk 1]
```diff
@@ -26,7 +26,7 @@ import fr.acinq.eclair.ApiTypes.ChannelNotFound
 import fr.acinq.eclair.TestConstants._
 import fr.acinq.eclair.blockchain.DummyOnChainWallet
 import fr.acinq.eclair.blockchain.fee.{FeeratePerByte, FeeratePerKw}
-import fr.acinq.eclair.channel._
+import fr.acinq.eclair.channel.{Command, _}
```
[the remark this answers is not in the document]
Comment (btcontract) [reply eclair-core/src/test/scala/fr/acinq/eclair/EclairImplSpec.scala]: It's not, probably was an automatic change.

**Review by t-bast — APPROVED** [review]

**Pull request merged**